### ISSUE TITLE
`failwith` rules produce incorrect warning when throwing in `Option.defaultWith`

### ISSUE THREAD

**Issue by peterhirn** [fsprojects/FSharpLint#539]
### Description

`failwith` rules produce incorrect warning when throwing in `Option.defaultWith`.

### Repro steps

```fsharp
let failIfNone (opt: 'a option) : 'a =
    opt
    |> Option.defaultWith (fun _ -> failwith "A unique error message")
```

### Actual behavior

```
Bad usage of failwith: Consider using unique error messages with failwith.
Error on line 3 starting at column 36
    |> Option.defaultWith (fun _ -> failwith "A unique error message")
```

### Known workarounds

Add message argument

```fsharp
let failIfNone msg (opt: 'a option) : 'a =
    opt |> Option.defaultWith (fun _ -> failwith msg)
```

### Related information

* Windows 10
* 0.21.1
* dotnet 6.0.201

**Comment by knocte** [comment]
Hey @peterhirn thanks for reporting this bug.

Are you sure this is related to Option.defaultWith? The error "Consider using unique error messages with failwith." is supposed to happen when you have more than one failwith call with the same error message. How many failwith calls with "1" do you have in your codebase?

**Comment by peterhirn** [comment]
Hi. I'm not sure if this only happens with `defaultWith` but I ran lint against a single file containing only this function to reproduce the problem. So codebase = this function, error message is unique.

**Comment by peterhirn** [comment]
Rule also applies to every `extern` function, eg.

```
Bad usage of failwith: Consider using unique error messages with failwith.
Error on line 18 starting at column 4
    extern int randombytes_uniform(int upperBound)
    ^                                             
```

**Comment by knocte** [comment]
> Rule also applies to every extern function, eg.

We reproduce the defaultWith issue, but not this one.

**Comment by peterhirn** [comment]
It seems to only happen when there are multiple `extern` functions in a file

```fsharp
open System.Runtime.InteropServices

[<DllImport("libsodium", CallingConvention = CallingConvention.Cdecl)>]
extern int crypto_pwhash(byte [] buffer, int64 bufferLength, byte [] password, int64 passwordLength, byte [] salt, int64 opsLimit, int memLimit, int alg)

[<DllImport("libsodium", CallingConvention = CallingConvention.Cdecl)>]
extern int randombytes_uniform(int upperBound)
```

**Comment by peterhirn** [comment]
I found another problem with this rule when throwing in a if/else

```fsharp
if false then
    ()
else
    failwith "Unique error message"
```

```
Bad usage of failwith: Consider using unique error messages with failwith.
Error on line 4 starting at column 4
    failwith "Unique error message"
    ^
```

**Comment by knocte** [comment]
@peterhirn we fixed the first case you raised here. I'm going to make a new release, please test it and if you see more false positives, raise each case as one issue. Thanks
